src/dsl/tokenizer.rs

```diff
@@ -83,6 +83,51 @@ impl<'input> Tokenizer<'input> {
 
         Some(Ok(self.cache.as_ref().unwrap()))
     }
+
+    pub fn can_get_idx(&mut self) -> Option<Result<bool, ParseError>> {
+        self.peek().map(|res| {
+            res.map(|token| match &token.kind {
+                TokenKind::Key(k) => k
+                    .chars()
+                    .next()
+                    .map(|c| c.is_ascii_digit())
+                    .unwrap_or(false),
+                _ => false,
+            })
+        })
+    }
+
+    pub fn get_idx(&mut self) -> usize {
+        let token = self.next().unwrap().unwrap();
+
+        match token.kind {
+            TokenKind::Key(k) => {
+                let mut idx = 0;
+
+                for (i, c) in k.char_indices() {
+                    if !c.is_ascii_digit() {
+                        break;
+                    } else {
+                        idx = i;
+                    }
+                }
+
+                let mut k = k;
+
+                let rest = k.split_off(idx + 1);
+
+                if !rest.is_empty() {
+                    self.cache = Some(Token {
+                        pos: token.pos,
+                        kind: TokenKind::Key(rest),
+                    });
+                }
+
+                k[..idx + 1].parse().unwrap()
+            }
+            _ => unreachable!(),
```

Comment:
Need to add a few words of explanation why it is unreachable

Reply:
it is unreachable because user should never call this fn without calling `can_get_idx`. I'll add comment about this

Reply:
also will convert this to panic